src/braket/circuits/circuit.py

```diff
@@ -438,6 +440,136 @@ def add_circuit(
 
         return self
 
+    def add_noise(
+        self,
+        noise: Noise,
+        target_gates: Union[str, Iterable[str]] = None,
+        target_qubits: QubitSetInput = None,
+        target_times: Union[int, Iterable[int]] = None,
+        insert_strategy: str = "inclusive",
+    ) -> Circuit:
+        """
+        Add the provided `noise` to gates, qubits and time specified by `target_gates`,
+         `target_qubits` and `target_times`. When `target_gates`=None, `noise` is added
```

Comment:
I'd simplify this: "For any parameter that is None, that specification is ignored (e.g. if 'target_gates' and 'target_qubits' are None then the noise is added to all qubits at target_times). If 'target_gates', 'target_qubits', and 'target_times' are all None, then 'noise' is added to every qubit at every moment."